cmd/update-rke2-charts/main.go

```diff
@@ -0,0 +1,300 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"k8s.io/helm/pkg/chartutil"
+)
+
+var (
+	name    string
+	version string
+	gitSHA  string
+)
+
+const (
+	envCharts   = "CHARTS"
+	filePackage = "package.yaml"
+	fileChart   = "Chart.yaml"
+	fileValues  = "values.yaml"
+)
+
+const usage = `Version: %s
+%[2]s - rke2-charts version references update
+
+Usage: %[2]s [options] <chart name> [<version field>=<version value>]
+Options:
+  -h		show this help message
+  -c dir	rke2-charts directory (defaults: $CHARTS value or "rke2-charts")
+  -i		write changes into their respective files
+  -p		print resulting yaml files on STDOUT
+  -v		print version
+`
+
+func main() {
+	flag.Usage = func() {
+		w := os.Stderr
+		for _, arg := range os.Args {
+			if arg == "-h" {
+				w = os.Stdout
+				break
+			}
+		}
+		fmt.Fprintf(w, usage, version, name)
+	}
+
+	var (
+		vers, inPlace, printOut bool
+		chartsDir               string
+	)
+
+	chartDefault := "rke2-charts"
+	if c, ok := os.LookupEnv(envCharts); ok {
+		chartDefault = c
+	}
+
+	flag.BoolVar(&vers, "v", false, "print version")
+	flag.BoolVar(&inPlace, "i", false, "write changes into their respective files")
+	flag.BoolVar(&printOut, "p", false, "print resulting yaml files on STDOUT")
+	flag.StringVar(&chartsDir, "c", chartDefault, "rke2-charts directory")
+	flag.Parse()
+
+	if vers {
+		fmt.Fprintf(os.Stdout, "version: %s - git sha: %s\n", version, gitSHA)
+		return
+	}
+
+	args := flag.Args()
+
+	if len(args) == 0 {
+		exitErr("no chart provided")
+	}
+
+	chartName := flag.Arg(0)
+	chartPath := filepath.Join(chartsDir, "packages", chartName)
+
+	overrides := make(map[string]string, len(args))
+	for _, v := range args[1:] {
+		a := strings.Split(v, "=")
+
+		if len(a) < 2 {
+			continue
+		}
+
+		overrides[a[0]] = a[1]
+	}
+
+	err := filepath.Walk(chartPath, func(path string, info fs.FileInfo, err error) error {
```

Comment:
I agree that it feels that way. Unfortunately, I found many _packages_ that does not follow that pattern like: 

- [rke2-Calico](https://github.com/rancher/rke2-charts/tree/main-source/packages/rke2-calico/templates/crd-template)
- [harvester-csi-driver](https://github.com/rancher/rke2-charts/tree/main-source/packages/harvester-csi-driver)
- Others whos package.yaml url is not `local`

So, I had to rely on a `Walk`. This can be simplified in the way you suggest if we allow the tool to ignore any package that does not follow said file layout.